platform/ios/src/UIColor+MGLAdditions.mm

```diff
@@ -12,6 +12,14 @@ @implementation UIColor (MGLAdditions)
     return { static_cast<float>(r*a), static_cast<float>(g*a), static_cast<float>(b*a), static_cast<float>(a) };
 }
 
+// Intended to be used when providing colors on the basis of an NSExpression
+- (mbgl::Color)mgl_colorForPremultipliedValue
```

Comment:
The two remaining code paths using `-mgl_color` are for styling annotations:

* `-[MGLMapView fillColorForPolygonAnnotation:]` → `-[MGLPolygon annotationObjectWithDelegate:]` → `FillAnnotationImpl::updateStyle()` → `FillLayer::setFillColor()`
* `-[MGLMapView strokeColorForShapeAnnotation:]` → `-[MGLPolyline annotationObjectWithDelegate:]` → `LineAnnotationImpl::updateStyle()` → `LineLayer::setLineColor()`

Annotations are backed by style layers under the hood, but the colors are expressed as raw `mbgl::PropertyValue<mbgl::Color>`s rather than expressions. Is the expectation of premultiplied values specific to NSExpression or paint property values in general?

In other words, do shape annotations have the correct colors when the application specifies the same UIColors as they would in style layer properties?

/ref https://github.com/mapbox/mapbox-gl-native/pull/16223#discussion_r383694089

Reply:
> Annotations are backed by style layers under the hood, but the colors are expressed as raw mbgl::PropertyValue<mbgl::Color>s rather than expressions. Is the expectation of premultiplied values specific to NSExpression or paint property values in general?

Expressions expect non-premultiplied, and `mbgl::Color` premultiplied. According to this comment https://github.com/mapbox/mapbox-gl-native/pull/16153#issuecomment-580171975

Reply:
mapbox/mapbox-gl-native-ios#454